rust/src/sip/sip.rs

```diff
@@ -112,6 +109,15 @@ impl SIPState {
         }
     }
 
+    fn append_request(&mut self, input: &[u8], request: Request) {
```

Comment:
naming is a bit weird. I would expect something to be appended to an existing request, but we're actually setting up a TX. Please reword

Reply:
renamed `append_request` and `append_response` to `build_tx_request` and `build_tx_response`.